lib/secretsscanner/authorizedkeys/authorized_keys.go

```diff
@@ -197,30 +201,51 @@ func (w *Watcher) start(ctx context.Context) error {
 	// maxReSendInterval is the maximum interval to re-send the authorized keys report
 	// to the cluster in case of no changes.
 	const maxReSendInterval = accessgraph.AuthorizedKeyDefaultKeyTTL - 20*time.Minute
-	timer := w.clock.NewTimer(jitterFunc(maxReSendInterval))
-	defer timer.Stop()
-	for {
+	expirationTimer := w.clock.NewTimer(jitterFunc(maxReSendInterval))
+	defer expirationTimer.Stop()
 
-		err := w.fetchAndReportAuthorizedKeys(ctx, fileWatcher)
-		interval := maxReSendInterval
-		if err != nil {
-			w.logger.WarnContext(ctx, "Failed to report authorized keys", "error", err)
-			interval = maxInitialDelay
-		}
+	// monitorTimer is the timer to monitor existing authorized keys.
+	const monitorTimerInterval = 3 * time.Minute
+	monitorTimer := w.clock.NewTimer(jitterFunc(monitorTimerInterval))
+	defer monitorTimer.Stop()
 
+	resetTimer := func(timer clockwork.Timer, interval time.Duration) {
 		if !timer.Stop() {
 			select {
 			case <-timer.Chan():
 			default:
 			}
 		}
 		timer.Reset(jitterFunc(interval))
+	}
+
+	var requiresReportToExtendTTL bool
+	for {
+
+		err := w.fetchAndReportAuthorizedKeys(ctx, fileWatcher, requiresReportToExtendTTL)
+		expirationTimerInterval := maxReSendInterval
+		if err != nil && !errors.Is(err, errKeysNotReported) {
+			w.logger.WarnContext(ctx, "Failed to report authorized keys", "error", err)
+			expirationTimerInterval = maxInitialDelay
+		}
+
+		// reset the mandatory report flag.
+		requiresReportToExtendTTL = false
+
+		// If the keys were reported, reset the expiration timer.
+		if !errors.Is(err, errKeysNotReported) {
+			resetTimer(expirationTimer, expirationTimerInterval)
```

Comment:
If there's an actual error (not `errKeysNotReported`) this will reset the expiration timer no? This might be the desired behavior, but I wasn't sure based on the code comment above.